src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveCMarshal.NativeAot.cs

```diff
@@ -51,20 +53,97 @@ internal static void ThrowPendingExceptionObject()
             }
         }
 
+        [UnmanagedCallersOnly]
+        static bool TryGetTaggedMemory(IntPtr pObj, IntPtr* tagged)
+        {
+            // We are paused in the GC, so this is safe.
+            object obj = Unsafe.AsRef<object>((void*)&pObj);
+
+            if (!s_objects.TryGetValue(obj, out ObjcTrackingInformation? info))
```

Comment:
I do not think `ConditionWeakTable` was intentionally designed so that `TryGetValue` could be run from a GC callout. From examining the implementation, I think it will work, so it should be ok. 

The alternative is to store this information in the syncblock, which is a lot more involved change, so probably can wait until it is really needed.